t/167-server-rewite.t

```diff
@@ -0,0 +1,223 @@
+--- no_error_log
+[error]
+
+=== TEST 2: server_rewrite_by_lua_block in server
+--- config
+	server_rewrite_by_lua_block {
+		ngx.log(ngx.INFO, "server_rewrite_by_lua_block in server")
+	}
+   	location /lua {
+       	content_by_lua_block {
+       	    ngx.say("OK")
+       	}
+   	}
+--- request
+GET /lua
+--- response_body
+OK
+--- error_log
+server_rewrite_by_lua_block in server
+--- no_error_log
+[error]
+
+
+=== TEST 3: server_rewrite_by_lua_block overwrite by server
+--- http_config
+	server_rewrite_by_lua_block {
+		ngx.log(ngx.INFO, "server_rewrite_by_lua_block in http")
+	}
+--- config
+	server_rewrite_by_lua_block {
+		ngx.log(ngx.INFO, "server_rewrite_by_lua_block in server")
+	}
+   	location /lua {
+       	content_by_lua_block {
+       	    ngx.say("OK")
+       	}
+   	}
+--- request
+GET /lua
+--- response_body
+OK
+--- error_log
+server_rewrite_by_lua_block in server
+--- no_error_log
+[error]
+
+
+=== TEST 4: sleep
+--- config
+	server_rewrite_by_lua_block {
+		ngx.sleep(0.001)
+		ngx.log(ngx.INFO, "server_rewrite_by_lua_block in server")
+	}
+   	location /lua {
+       	content_by_lua_block {
+       	    ngx.say("OK")
+       	}
+   	}
+--- request
+GET /lua
+--- response_body
+OK
+--- error_log
+server_rewrite_by_lua_block in server
+--- no_error_log
+[error]
+
+
+=== TEST 5: ngx.exit(ngx.OK)
+--- config
+	server_rewrite_by_lua_block {
+		ngx.log(ngx.INFO, "ngx.exit")
+        ngx.exit(ngx.OK)
+	}
+   	location /lua {
+       	content_by_lua_block {
+           ngx.say("OK")
+       	}
+   	}
+--- request
+GET /lua
+--- response_body
+OK
+--- error_log
+ngx.exit
+--- no_error_log
+[error]
+
+
+=== TEST 6: ngx.exit(503)
+--- config
+	server_rewrite_by_lua_block {
+        ngx.exit(503)
+	}
+   	location /lua {
+       	content_by_lua_block {
+			ngx.log(ngx.ERR, "content_by_lua")
+        	ngx.say("OK")
+       	}
+   	}
+--- request
+GET /lua
+--- error_code: 503
+--- no_error_log
+[error]
+
+
+=== TEST 7: subrequests
+--- config
+	server_rewrite_by_lua_block {
+		ngx.log(ngx.INFO, "is_subrequest:", ngx.is_subrequest)
+	}
+
+   	location /lua {
+       	content_by_lua_block {
+			local res = ngx.location.capture("/sub")
+			ngx.print(res.body)
+		}
+   	}
+
+	location /sub {
+       	content_by_lua_block {
+           	ngx.say("OK")
+       	}
+   	}
+
+--- request
+GET /lua
+--- response_body
+OK
+--- error_log
+is_subrequest:true
+is_subrequest:false
+--- no_error_log
+[error]
+
+
+=== TEST 8: rewrite by ngx_http_rewrite_module
+--- config
+	server_rewrite_by_lua_block {
+		ngx.log(ngx.INFO, "uri is ", ngx.var.uri)
+	}
+
+	rewrite ^ /re;
+
+	location /re {
+       	content_by_lua_block {
+           	ngx.say("RE")
+       	}
+   	}
+
+	location /ok {
+       	content_by_lua_block {
+           	ngx.say("OK")
+       	}
+	}
+
+--- request
+GET /lua
+--- response_body
+RE
+--- error_log
+uri is /lua
+--- no_error_log
+[error]
+
+
+=== TEST 9: exec
+--- config
+	server_rewrite_by_lua_block {
```

Comment:
You should add a test case when using both `server_rewrite_by_lua*` and `rewrite_by_lua*`? Both in the same scope or in different scopes.

Reply:
TEST 10